pubtools/pulplib/_impl/client/client.py

```diff
@@ -165,20 +165,45 @@ def search_repository(self, criteria=None):
                 Each page will contain a collection of
                 :class:`~pubtools.pulplib.Repository` objects.
         """
+        search_options = {"distributors": True}
+        return self._search(
+            Repository, "repositories", criteria=criteria, search_options=search_options
+        )
+
+    def search_distributor(self, criteria=None):
+        """Search the distributors matching the given criteria
+
+        Args:
+            criteria (:class:`~pubtools.pulplib.Criteria`)
+                A criteria object used for this search.
+                If None, search for all repositories.
+
+        Returns:
+            Future[:class:`~pubtools.pulplib.Page`]
+                A future representing the first page of results.
+
+                Each page will contain a collection of
+                :class:`~pubtools.pulplib.Distributor` objects.
+        """
+        return self._search(Distributor, "distributors", criteria=criteria)
+
+    def _search(self, return_type, resource_type, criteria=None, search_options=None):
         pulp_crit = {
             "skip": 0,
             "limit": self._PAGE_SIZE,
-            "filters": filters_for_criteria(criteria, Repository),
+            "filters": filters_for_criteria(criteria, return_type),
         }
-        search = {"criteria": pulp_crit, "distributors": True}
+        search = {"criteria": pulp_crit}
+        if search_options:
+            search.update(search_options)
```

Comment:
A trivial suggestion: in cases like this you can consider instead dropping the "if" and writing

```suggestion
      search.update(search_options or {})
```

The difference being that you're not creating a new branch of code counted separately by coverage tools.  I would usually but not always count that as a benefit.